### ISSUE TITLE
Armory can't be enabled due to Shadow & Light, which is no longer installed

### ISSUE THREAD

**Issue by Quaantt** [Toxicom/toxiui#83]
I had installed S&L to toy with it, but I've since disabled and uninstalled. Toxi is stuck thinking S&L Armory is enabled.

**Comment by Quaantt** [comment]
Reinstalled S&L to troubleshoot. I get this error when both S&L and Toxi are enabled:

ElvUI_ToxiUI/Core/Requirements.lua:44: attempt to index field 'enchant' (a nil value)
[string "@ElvUI_ToxiUI/Core/Requirements.lua"]:44: in function `CheckRequirements'
[string "@ElvUI_ToxiUI/Core/Requirements.lua"]:7: in function `HasRequirements'
[string "@ElvUI_ToxiUI/ModulesDeconstruct.lua"]:795: in function `callback'
[string "@ElvUI_ToxiUI/Core/Functions/Event.lua"]:285: in function `ContinueOutOfCombat'
[string "@ElvUI_ToxiUI/ModulesDeconstruct.lua"]:790: in function <...eElvUI_ToxiUI/ModulesDeconstruct.lua:786>
[string "=(tail call)"]: ?
[string "@ElvUI_ToxiUI/Core/Functions/Event.lua"]:252: in function <ElvUI_ToxiUI/Core/Functions/Event.lua:250>
[string "=[C]"]: ?
[string "@ElvUI_ToxiUI/Core/Functions/Event.lua"]:170: in function <ElvUI_ToxiUI/Core/Functions/Event.lua:169>
[string "=[C]"]: ?
[string "@ElvUI_ToxiUI/Core/Functions/Event.lua"]:173: in function `TriggerEvent'
[string "@ElvUI_ToxiUI/Core/Core.lua"]:132: in function `callback'
[string "@ElvUI_ToxiUI/Core/Functions/Event.lua"]:285: in function `ContinueOutOfCombat'
[string "@ElvUI_ToxiUI/Core/Core.lua"]:130: in function `callback'
[string "@ElvUI_ToxiUI/Core/Functions/Event.lua"]:312: in function `ContinueAfter'
[string "@ElvUI_ToxiUI/Core/Functions/Event.lua"]:351: in function `ContinueAfterElvUIUpdate'
[string "@ElvUI_ToxiUI/Core/Core.lua"]:101: in function <ElvUI_ToxiUI/Core/Core.lua:99>
[string "@ElvUI_ToxiUI/Core/Functions/Event.lua"]:379: in function <ElvUI_ToxiUI/Core/Functions/Event.lua:374>
[string "=[C]"]: ?
[string "@ElvUI_ToxiUI/Core/Functions/Event.lua"]:170: in function <ElvUI_ToxiUI/Core/Functions/Event.lua:169>
[string "=[C]"]: ?
[string "@ElvUI_ToxiUI/Core/Functions/Event.lua"]:173: in function `TriggerEvent'
[string "@ElvUI_ToxiUI/Core/Functions/Event.lua"]:388: in function <ElvUI_ToxiUI/Core/Functions/Event.lua:387>

Locals:
self = <table> {
 SetDefaultModuleLibraries = <function> defined @ElvUI_Libraries/Core/Ace3/AceAddon-3.0/AceAddon-3.0.lua:363
 IsRetail = true
 DevTag = ""
 EnableModule = <function> defined @ElvUI_Libraries/Core/Ace3/AceAddon-3.0/AceAddon-3.0.lua:328
 modules = <table> {
 }
 CancelTimer = <function> defined @ElvUI_Libraries/Core/Ace3/AceTimer-3.0/AceTimer-3.0.lua:147
 IterateEmbeds = <function> defined @ElvUI_Libraries/Core/Ace3/AceAddon-3.0/AceAddon-3.0.lua:438
 SecureHookScript = <function> defined @ElvUI_Libraries/Core/Ace3/AceHook-3.0/AceHook-3.0.lua:409
 LogTrace = <function> defined @ElvUI_ToxiUI/Core/Functions/Log.lua:148
 DevRelease = false
 MetaFlavor = "Mainline"
 HandleDevExports = <function> defined @ElvUI_ToxiUI/Core/Commands.lua:52
 Printf = <function> defined @ElvUI_Libraries/Core/Ace3/AceConsole-3.0/AceConsole-3.0.lua:65
 SetDefaultModulePrototype = <function> defined @ElvUI_Libraries/Core/Ace3/AceAddon-3.0/AceAddon-3.0.lua:405
 HasRequirements = <function> defined @ElvUI_ToxiUI/Core/Requirements.lua:6
 defaultModuleState = true
 IsEnabled = <function> defined @ElvUI_Libraries/Core/Ace3/AceAddon-3.0/AceAddon-3.0.lua:447
 ScheduleTimer = <function> defined @ElvUI_Libraries/Core/Ace3/AceTimer-3.0/AceTimer-3.0.lua:94
 DisableModule = <function> defined @ElvUI_Libraries/Core/Ace3/AceAddon-3.0/AceAddon-3.0.lua:346
 ThrowError = <function> defined @ElvUI_ToxiUI/Core/Functions/Log.lua:83
 RegisterModule = <function> defined @ElvUI_ToxiUI/Core/Core.lua:23
 RegisteredModules = <table> {
 }
 LogLevel = 3
 GetArgs = <function> defined @ElvUI_Libraries/Core/Ace3/AceConsole-3.0/AceConsole-3.0.lua:140
 hooks = <table> {
 }
 UpdateProfiles = <function> defined @ElvUI_ToxiUI/Core/Core.lua:27
 SetEnabledState = <function> defined @ElvUI_Libraries/Core/Ace3/AceAddon-3.0/AceAddon-3.0.lua:420
 Hook = <function> defined @ElvUI_Libraries/Core/Ace3/AceHook-3.0/AceHook-3.0.lua:273
 enabledState = true
 HandleDevProfiles = <function> defined @ElvUI_ToxiUI/Core/Commands.lua:6
 IterateModules = <function> defined @ElvUI_Libraries/Core/Ace3/AceAddon-3.0/AceAddon-3.0.lua:433
 Title = "|cffffffffToxi|r|cff18a8ffUI|r"
 RawHook = <function> defined @ElvUI_Libraries/Core/Ace3/AceHook-3.0/AceHook-3.0.lua:312
 isEnabled = false
 Enable = <function> defined @ElvUI_Libraries/Core/Ace3/AceAddon-3.0/AceAddon-3.0.lua:290
 RegisterChatCommand = <function> defined @ElvUI_Libraries/Core/Ace3/AceConsole-3.0/AceConsole-3.0.lua:81
 Changelog = <table> {
 }
 GetModule = <function> defined @ElvUI_Libraries/Core/Ace3/AceAddon-3.0/AceAddon-3.0.lua:206
 ExportProfile = <function> defined @ElvUI_ToxiUI/Core/Export.lua:100
 IsClassic = false
 defaultModuleLibraries = <table> {
 }
 TimeLeft = <function> defined @ElvUI_Libraries/Core/Ace3/AceTimer-3.0/AceTimer-3.0.lua:172
 GitHash = "v6.5.0"
 UnregisterChatCommand = <function> defined @ElvUI_Libraries/Core/Ace3/AceConsole-3.0/AceConsole-3.0.lua:107
 initializedSafe = true
 ScheduleRepeatingTimer = <function> defined @ElvUI_Libraries/Core/Ace3/AceTimer-3.0/AceTimer-3.0.lua:129
 CancelAllTimers = <function> defined @ElvUI_Libraries/Core/Ace3/AceTimer-3.0/AceTimer-3.0.lua:160
 initialized = true
 RawHookScript = <function> defined @ElvUI_Libraries/Core/Ace3/AceHook-3.0/AceHook-3.0.lua:394
 GetName = <function> defined @ElvUI_Libraries/Core/Ace3/AceAddon-3.0/AceAddon-3.0.lua:275
 LogInfo = <function> defined @ElvUI_ToxiUI/Core/Functions/Log.lua:138
 SetupDevRelease = <function> defined @ElvUI_ToxiUI/Core/Core.lua:40
 LogWarning = <function> def

**Comment by Toxicom** [comment]
Can you attach a `/tx status` image?

It should not error out on that line cause it does first check if S&L is enabled at all:
```lua
if F.IsAddOnEnabled("ElvUI_SLE") and E.private.sle and (E.private.sle.professions.deconButton.enable or E.private.sle.professions.enchant.enchScroll) then
```

**Comment by Toxicom** [comment]
I do see an issue with armory checking though, basically we check if the S&L armory is disabled in the database and we do not check for enabled addon at all, so you could have S&L uninstalled but the database would still show it as enabled. I'll fix that

**Comment by Toxicom** [comment]
@Quaantt I've released a `v6.5.1-beta.7` build that includes some improvements for this issue. Could you please test with that and let me know if it resolves your issue?

edit: https://github.com/Toxicom/toxiui/releases/tag/v6.5.1-beta.7

**Comment by Quaantt** [comment]
I deleted all ElvUI files and did kind of a reset on my UI and that has solved this exact issue. I can still install that beta and run it if you'd like.
